kong/clustering/data_plane.lua

```diff
@@ -389,6 +388,20 @@ function _M:communicate(premature)
     ngx_log(ngx_ERR, _log_prefix, perr, log_suffix)
   end
 
+  -- the config thread might be holding a lock if it's in the middle of an
+  -- update, so we need to give it a chance to terminate gracefully
+  config_exit = true
+  ok, err, perr = ngx.thread.wait(config_thread)
+
+  ngx.thread.kill(config_thread)
```

Comment:
Is this needed after `ngx.thread.wait` already returns? `config_thread` should be dead either way at this point.

Reply:
Ah, you're right. I'll remove this.